torch/distributed/_tensor/placement_types.py

```diff
@@ -40,13 +40,9 @@ def _split_tensor(
         with_padding: bool = True,
         contiguous: bool = True,
     ) -> Tuple[List[torch.Tensor], List[int]]:
-        # NOTE: For with_padding option, we pad the tensor on each rank before calling
-        # the collectives (i.e. scatter/all_gather, etc.). This is because for gloo
-        # backend, it does not support uneven collectives, nccl supports some, but
-        # it might be slow compared to even size collective, we need to pad tensor
-        # before really calling the collective, and unpad/narrow it afterwards
-        # TODO: consider if we should remove this logic once ProcessGroupGloo
-        # support uneven list, and collective performance on par
+        # NOTE: For with_padding option, we pad the tensor on the last few rank
```

Comment:
"on the last few rank(s)" maybe? Not sure on this because the behavior is not easy to describe. 

Maybe we can mention that this function splits tensor uses `torch.chunk` while keep the result tensor size to be `num_chunks` by padding the empty or smaller tensor.